### PR TITLE
New methods for handling token for guest users to prevent CSRF

### PR DESCRIPTION
**Pull request by kpodemski**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | Currently getToken doesn't allow to prevent CSRF for actions related to guest users |
| Type? | improvement |
| Category? | CO |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? | n/a |
| How to test? | n/a |

#### Important guidelines
- Make sure [your local branch is up to date](https://help.github.com/articles/syncing-a-fork/) before commiting your changes!
- Your code MUST respect [our Coding Standards](http://doc.prestashop.com/display/PS16/Coding+Standards) (for code written in PHP, JavaScript, HTML/CSS/Smarty/Twig, SQL)!
- Your commit name MUST respect our [naming convention](http://doc.prestashop.com/display/PS16/How+to+write+a+commit+message)!

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/5862)

### CODE REVIEW

**Comment by mickaelandrieu** [comment]
Hi @kpodemski, nice idea!

what do you think @rGaillard  ?

Mickaël

**Comment by firstred** [comment]
Status: Code reviewed

**Comment by mickaelandrieu** [comment]
@firstred nice try ^^

**Comment by aleeks** [comment]
re-ping @rGaillard

**Comment by rGaillard** [comment]
Hey @kpodemski,

Great job!
Maybe that we can directly add these changes in the getToken() and isValidToken() instead of adding two new methods? How do you think?

**Comment by aleeks** [comment]
Hello @kpodemski 
What do you think about @rGaillard response ?

File: classes/controller/FrontController.php [hunk 1]
```diff
@@ -920,6 +920,21 @@ public function isTokenValid()
         return (strcasecmp(Tools::getToken(false), Tools::getValue('token')) == 0);
     }
 
+    /**
+     * Checks if guest token is valid
+     *
+     * @since 1.6.1.7
```
Comment (mickaelandrieu) [inline classes/controller/FrontController.php:926]: since 1.7.4? :)

**Comment by mickaelandrieu** [comment]
@kpodemski do you think this still have value in `1.7.x` branch? We are migrating the BO the Symfony and the FO have been totally reworked.

Feel free to re-open it if it does.

**Pull request closed without merging**